R/tmb.R

```diff
@@ -426,11 +442,13 @@ h_mmrm_tmb_fit <- function(tmb_object,
 #' system.time(result <- h_mmrm_tmb(formula, data))
 h_mmrm_tmb <- function(formula,
                        data,
+                       weights,
                        reml = TRUE,
                        control = h_mmrm_tmb_control()) {
   formula_parts <- h_mmrm_tmb_formula_parts(formula)
   assert_class(control, "mmrm_tmb_control")
-  tmb_data <- h_mmrm_tmb_data(formula_parts, data, reml, accept_singular = control$accept_singular)
+  assert_vector(weights)
```

Comment:
I would assert this here 
```suggestion
  assert_numeric(weights, any.missing = FALSE)
  assert_true(all(weights > 0))
```